ArdeleanuStefan_Task5/include/DynamicArray.hpp

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include <cstring>
+#include <iostream>
+#include <string>
+#include <vector>
+
+template <typename T>
+class DynamicArray
+{
+private:
+    T *dynamicArray;
+    size_t m_size;
+
+public:
+    DynamicArray() : m_size{0}, dynamicArray{nullptr}
+    {
+    }
+
+    DynamicArray(const std::vector<T> &inputs) : DynamicArray()
+    {
```

Comment:
Please implement the methods outside the class definition in order to separate the interface from the implementation.